src/cargo-kani/src/call_cargo.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::{bail, Context, Result};
+use std::ffi::OsString;
+use std::path::PathBuf;
+use std::process::Command;
+
+use crate::context::KaniContext;
+
+impl KaniContext {
```

Comment:
This is a general comment and I think we should address in a separate PR, but I would prefer breaking some of this logic down into more than one struct. I think it would be cleaner if we had smaller structs. We are using self a lot, and that could possible be broken down by calling other objects.